### ISSUE TITLE
Considerable features

### ISSUE THREAD

**Issue by ysaakpr** [Azure/brigade#652]
1. Provision to run only one build at a time for a project.
In case if multiple triggers like git push happens, multiple builds are getting triggered, which should wait for one to complete before starting (and Possibly can allow merging of waiting events for a project)

**Comment by Strandedpirate** [comment]
+1 I've coded this exact customization on top of brigade for my company. I'm thinking of open sourcing it but it would be best if it was baked into brigade.

**Comment by radu-matei** [comment]
Closing in favor of #172, which offers a bit more context.
Feel free to comment over there, or reopen if this is a separate request.

Thanks!
